src/shared/cpu-set-util.c

```diff
@@ -287,3 +295,103 @@ int cpu_set_from_dbus(const uint8_t *bits, size_t size, CPUSet *set) {
         s = (CPUSet) {};
         return 0;
 }
+
+static int numa_policy_to_mempolicy(const NUMAPolicy *policy, unsigned long *maxnode, unsigned long **nodes) {
+        unsigned node, bits, ulong_bits;
+        _cleanup_free_ unsigned long *out = NULL;
+
+        assert(policy);
+        assert(maxnode);
+        assert(nodes);
+
+        /* Take a shortcut here if policy is set to default or local since kernel requires that we pass NULL as nodemask */
+        if (policy->type == NUMA_POLICY_DEFAULT || policy->type == NUMA_POLICY_LOCAL)
+                goto out;
+
+        bits = policy->nodes.allocated * 8;
+        ulong_bits = sizeof(unsigned long) * 8;
```

Comment:
Not sure I grok this.

Reply:
You're calling `new0` with `bits / ulong_bits` rounded up. I think it'd be easier to use `DIV_ROUND_UP(bits, ulong_bits)` instead of `(bits / ulong_bits) + (bits % ulong_bits  > 0)`.